### PR TITLE
Import CGMES area exchanges

### PR DESCRIPTION
**Pull request by MioRtia**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
New feature

**What is the current behavior?** *(You can also link to an open issue here)*
CGMES area exchanges are not imported

**What is the new behavior (if this is a feature change)?**
CGMES area exchanges are imported

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
No real breaking change but implementation of CgmesControlAreas and CgmesControlArea is mandatory if you want to use this version's CGMES import. Some features might also not work correctly if you do not implement the new methods of `Boundary`

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: cgmes/cgmes-extensions/src/test/resources/eurostag_cgmes_control_area.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            </iidm:ratioTapChanger>
+        </iidm:twoWindingsTransformer>
+    </iidm:substation>
+    <iidm:line id="NHV1_NHV2_1" r="3.0" x="33.0" g1="0.0" b1="1.93E-4" g2="0.0" b2="1.93E-4" bus1="NHV1" connectableBus1="NHV1" voltageLevelId1="VLHV1" bus2="NHV2" connectableBus2="NHV2" voltageLevelId2="VLHV2"/>
+    <iidm:line id="NHV1_NHV2_2" r="3.0" x="33.0" g1="0.0" b1="1.93E-4" g2="0.0" b2="1.93E-4" bus1="NHV1" connectableBus1="NHV1" voltageLevelId1="VLHV1" bus2="NHV2" connectableBus2="NHV2" voltageLevelId2="VLHV2"/>
+    <iidm:extension id="sim1">
+        <cca:cgmesControlAreas>
+            <cca:controlArea id="cgmesControlAreaId" name="cgmesControlAreaName" energyIdentCodeEic="energyIdentCodeEic" netInterchange="100.0">
+                <cca:terminal id="NHV1_NHV2_1" side="ONE"/>
+                <cca:boundary id="DL"/>
```
Comment (annetill) [inline cgmes/cgmes-extensions/src/test/resources/eurostag_cgmes_control_area.xml:48]: What is the convention here for `boundary` object ? It is always the terminal on boundary side?

Comment (MioRtia) [reply cgmes/cgmes-extensions/src/test/resources/eurostag_cgmes_control_area.xml:48]: yes (it is actually the boundary object)

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Boundary.java [hunk 2]
```diff
@@ -18,4 +18,12 @@
     double getP();
 
     double getQ();
+
+    default Branch.Side getSide() {
```
Comment (annetill) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Boundary.java:22]: You need this function to be able to build a terminal ref or something similar?

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineBoundaryImpl.java [hunk 3]
```diff
@@ -53,6 +51,21 @@ public double getQ() {
         return new SV(t.getP(), t.getQ(), getV(b), getAngle(b)).otherSideQ(parent);
     }
 
+    @Override
+    public Branch.Side getSide() {
```
Comment (annetill) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineBoundaryImpl.java:55]: Same question.

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Boundary.java (hunk 2 above)
Comment (MioRtia) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Boundary.java:22]: Boundary can refers to a TieLine's boundary, in which case the user should know which "side" they are looking at

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineBoundaryImpl.java (hunk 3 above)
Comment (MioRtia) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineBoundaryImpl.java:55]: For dangling line, the side is always null though...

File: cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CgmesControlArea.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public interface CgmesControlArea {
+
+    String getId();
+
+    String getName();
+
+    String getEnergyIdentCodeEic();
```
Comment (annetill) [inline cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CgmesControlArea.java:23]: Just a proposal: `EnergyIdentificationCodeEIC` to be more explicit.

**Review by marcosmc — APPROVED** [review]

**Review by zamarrenolm — APPROVED** [review]

**Review by annetill — APPROVED** [review]

**Pull request merged**